### ISSUE TITLE
💡Using Checkstyle with the agreed convention for the arrow-hand switch case notation.

### ISSUE THREAD

**Issue by zaidansani** [nus-cs2103-AY2425S1/forum#184]
As mentioned in #49, the preferred switch case convention with the arrow-hand notation is as such:
```Java
TaskType taskType = switch (data.charAt(0)) {
    case 'T' -> TaskType.TODO;
    case 'E' -> TaskType.EVENT;
    case 'D' -> TaskType.DEADLINE;
    default -> TaskType.UNKNOWN;
}; 
```

However, this goes against the indentation rule where cases are supposed to be indented by `0`. 
If the `@SuppressWarnings` annotation is attempted to be used, the script still flags for this build mistake.

Instead, use the inbuilt SuppressionCommentFilter:
```Java
<module name="SuppressionCommentFilter">
            <property name="offCommentFormat" value="CHECKSTYLE.OFF\: ([\w\|]+)"/>
            <property name="onCommentFormat" value="CHECKSTYLE.ON\: ([\w\|]+)"/>
            <property name="checkFormat" value="$1"/>
        </module>
```

and add the comments above and below: for example:

```Java
// CHECKSTYLE.OFF: Indentation
TaskType taskType = switch (data.charAt(0)) {
    case 'T' -> TaskType.TODO;
    case 'E' -> TaskType.EVENT;
    case 'D' -> TaskType.DEADLINE;
    default -> TaskType.UNKNOWN;
}; 
// CHECKSTYLE.ON: Indentation
```

which will disable the Indentation rule for the area between the two comments.

> Note: only use this if you're sure that your code matches the convention. In this scenario, a new Checkstyle custom rule is required since it does not recognise the arrow-notation, so this is a less taxing way of building without flagging the Checkstyle.

**Comment by SeeYangZhi** [comment]
> Instead, use the inbuilt SuppressionCommentFilter:
> 
> ```xml
> <module name="SuppressionCommentFilter">
>             <property name="offCommentFormat" value="CHECKSTYLE.OFF\: ([\w\|]+)"/>
>             <property name="onCommentFormat" value="CHECKSTYLE.ON\: ([\w\|]+)"/>
>             <property name="checkFormat" value="$1"/>
> </module>
> ```
> 

To clarify, the above should be added in `config/checkstyle/checkstyle.xml`

**Comment by zaidansani** [comment]
> > Instead, use the inbuilt SuppressionCommentFilter:
> > ```
> > <module name="SuppressionCommentFilter">
> >             <property name="offCommentFormat" value="CHECKSTYLE.OFF\: ([\w\|]+)"/>
> >             <property name="onCommentFormat" value="CHECKSTYLE.ON\: ([\w\|]+)"/>
> >             <property name="checkFormat" value="$1"/>
> > </module>
> > ```
> 
> To clarify, the above should be added in `config/checkstyle/checkstyle.xml`

Yep! I think it's already inbuilt though, if people are using the Checkstyle off AB3. Thanks for the clarification!

**Comment by yadobler** [comment]
If you're highly motivated you can compile a custom checkstyle module as [suggested by chatGPT here](https://chatgpt.com/share/66e3c7f5-97cc-800f-9b03-2335ef20d47f)

Then import it to your checkstyle.xml 

The main issue is that checkstyle does not discriminate between the different case statements

**Comment by zaidansani** [comment]
> If you're highly motivated you can compile a custom checkstyle module as [suggested by chatGPT here](https://chatgpt.com/share/66e3c7f5-97cc-800f-9b03-2335ef20d47f)
> 
> Then import it to your checkstyle.xml
> 
> The main issue is that checkstyle does not discriminate between the different case statements

Agreed. I personally realised I was only going to use the switch case there once, so decided against implementing this personally, but for those who utilise their switch cases more, it might be useful!
